test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -63,5 +71,31 @@ function ($, Q, Expect, $sandbox, DeviceFingerprint) {
       });
     });
 
+    it('fails if user agent is not defined', function (done) {
+      mockUserAgent(undefined);
+      mockIFrameMessages(true);
+      DeviceFingerprint.generateDeviceFingerprint('file://', $sandbox)
+        .then(function () {
+          done.fail('Fingerprint promise incorrectly resolved successful');
```

Comment:
Nit: Won't block on it, but I think a message like "Fingerprint promise should have been rejected" is more clear for all the ``done.fail()`` calls in this file.